jorvik/storage/isolation.py

```diff
@@ -0,0 +1,329 @@
+from typing import Callable
+import datetime
+
+from pyspark.sql import DataFrame, SparkSession, functions as F
+from pyspark.sql.streaming import StreamingQuery
+
+from delta.tables import DeltaTable
+
+from . import Storage
+
+class IsolatedStorage(Storage):
+    """
+    A storage wrapper that isolates data operations based on a context (e.g., branch name).
+    """
+
+    def __init__(self, storage: Storage, verbose: bool = False, isolation_provider: Callable = None):
+        """
+        Initialize IsolatedStorage.
+
+        Args:
+            storage (Storage): The underlying storage instance.
+            verbose (bool): Enable verbose logging.
+            isolation_provider (Callable): Function returning the isolation context (e.g., branch name).
+        """
+        self.storage = storage
+        self.verbose = verbose
+        self.isolation_provider = isolation_provider
+
+    def _get_isolation_context(self):
+        """
+        Get the current isolation context (e.g., branch name).
+
+        Returns:
+            str: The isolation context.
+
+        Raises:
+            ValueError: If the isolation context is invalid.
+        """
+
+        isolation = self.isolation_provider()
+
+        if isolation.endswith('/') or isolation.endswith('\\'):
+            raise ValueError("Isolation path name must not end with '/' or '\\'.")
+
+        if '|' in isolation:
+            raise ValueError("Isolation path name must not contain '|'.")
+
+        return isolation
+
+    def _configure_path(self, path: str, add_isolation: bool = True) -> str:
+        """
+        Configure the path based on the isolation context (e.g., branch name)
+        and the isolation container.
```

Comment:
not necessarily a container
```suggestion
        and the isolation path.
```